File: killrvideo-service-videocatalog/src/test/java/com/killrvideo/service/video/dao/LatestVideoRowMapperTest.java
package com.killrvideo.service.video.dao;

import com.datastax.oss.driver.api.core.cql.Row;
import com.killrvideo.service.video.dto.LatestVideo;
import org.junit.jupiter.api.Test;

import java.time.Instant;
import java.util.UUID;

import static com.killrvideo.dse.dto.AbstractVideo.COLUMN_NAME;
import static com.killrvideo.dse.dto.AbstractVideo.COLUMN_PREVIEW;
import static com.killrvideo.dse.dto.Video.*;
import static com.killrvideo.service.video.dto.LatestVideo.COLUMN_YYYYMMDD;
import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

class LatestVideoRowMapperTest {
    private LatestVideoRowMapper mapper = new LatestVideoRowMapper();

    @Test
    public void testMap() {
        Row row = mock(Row.class);

        String yyyymmdd = "20211231";
        UUID userid = UUID.randomUUID();
        UUID videoid = UUID.randomUUID();
        String name = "Game";
        String previewLocation = "previewUrl";
        Instant addedDate = Instant.now();

        when(row.getString(COLUMN_YYYYMMDD)).thenReturn(yyyymmdd);
        when(row.getUuid(COLUMN_USERID)).thenReturn(userid);
        when(row.getUuid(COLUMN_VIDEOID)).thenReturn(videoid);
        when(row.getString(COLUMN_NAME)).thenReturn(name);
        when(row.getString(COLUMN_PREVIEW)).thenReturn(previewLocation);
        when(row.getInstant(COLUMN_ADDED_DATE)).thenReturn(addedDate);

        LatestVideo result = mapper.map(row);
        assertEquals(yyyymmdd, result.getYyyymmdd());
        assertEquals(userid, result.getUserid());
        assertEquals(videoid, result.getVideoid());
        assertEquals(name, result.getName());
        assertEquals(previewLocation, result.getPreviewImageLocation());
        assertEquals(addedDate, result.getAddedDate());
    }
}

File: killrvideo-service-videocatalog/src/test/kotlin/com/killrvideo/service/video/dao/LatestVideoRowMapperTest.kt
package com.killrvideo.service.video.dao

import com.datastax.oss.driver.api.core.cql.Row
import com.killrvideo.service.video.dto.LatestVideo
import com.killrvideo.service.video.dto.Video
import io.mockk.every
import io.mockk.mockk
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.time.Instant
import java.util.*

internal class LatestVideoRowMapperTest {
    private val mapper = LatestVideoRowMapper()
    @Test
    fun testMap() {
        val yyyymmdd = "20211231"
        val userid = UUID.randomUUID()
        val videoid = UUID.randomUUID()
        val name = "Game"
        val previewLocation = "previewUrl"
        val addedDate = Instant.now()

        val row = mockk<Row>()
        every {row.getString(LatestVideo.COLUMN_YYYYMMDD)} returns yyyymmdd
        every {row.getUuid(Video.COLUMN_USERID)} returns userid
        every {row.getUuid(Video.COLUMN_VIDEOID)} returns videoid
        every {row.getString(Video.COLUMN_NAME)} returns name
        every {row.getString(Video.COLUMN_PREVIEW)} returns previewLocation
        every {row.getInstant(Video.COLUMN_ADDED_DATE)} returns addedDate

        val result = mapper.map(row)
        assertEquals(yyyymmdd, result.yyyymmdd)
        assertEquals(userid, result.userid)
        assertEquals(videoid, result.videoid)
        assertEquals(name, result.name)
        assertEquals(previewLocation, result.previewImageLocation)
        assertEquals(addedDate, result.addedDate)
    }
}
